### PR TITLE
service protocol 0 removed

### PR DESCRIPTION
**Pull request by rex-schilasky**
### Description
Service protocol v0 implementation removed (client and server side).

PR may conflict with #1842

### CODE REVIEW

File: ecal/core/src/service/ecal_service_server_impl.cpp [hunk 1]
```diff
@@ -331,8 +312,8 @@ namespace eCAL
     service.uname       = Process::GetUnitName();
     service.sname       = m_service_name;
 
-    service.tcp_port_v0 = server_tcp_port_v0;
-    service.tcp_port_v1 = server_tcp_port_v1;
+    service.tcp_port_v0 = 0;
```
Comment (KerstinKeller) [inline ecal/core/src/service/ecal_service_server_impl.cpp:315]: Is this the internal API?  should it be removed there, too?

File: ecal/service/ecal_service/src/client_session.cpp [hunk 2]
```diff
@@ -76,19 +75,13 @@ namespace eCAL
     }
 
     ClientSession::ClientSession(const std::shared_ptr<asio::io_context>&                   io_context
-                                , std::uint8_t                                              protocol_version
+                                , std::uint8_t                                              /*protocol_version*/
```
Comment (KerstinKeller) [inline ecal/service/ecal_service/src/client_session.cpp:78]: doesn't it make sense to remove the variable from the constructor?

File: ecal/core/src/service/ecal_service_server_impl.cpp (hunk 1 above)
Comment (rex-schilasky) [reply ecal/core/src/service/ecal_service_server_impl.cpp:315]: This is part of the eCAL service registration sample. We could remove this from the registration sample as well but we have to ensure that it is communicated as 0.

**Review by KerstinKeller — COMMENTED** [review]
In general very good that we can remove so much code!
Just see the two remarks that I made...

File: ecal/core/src/service/ecal_service_server_impl.cpp (hunk 1 above)
Comment (KerstinKeller) [reply ecal/core/src/service/ecal_service_server_impl.cpp:315]: Ok, maybe a separate PR for that, or at least we add it to the TODO list.

File: ecal/service/ecal_service/src/client_session.cpp (hunk 2 above)
Comment (rex-schilasky) [reply ecal/service/ecal_service/src/client_session.cpp:78]: This is the service library. Not sure if one day this library will introduce a new protocol again. This "protocol_version" is exiting all over the library implementation ..

**Review by KerstinKeller — APPROVED** [review]

**Pull request merged**